fix: add swap fee estimate approximation

kathaypacific opened the pull request:
### Description

This PR adds a fee estimate to swaps so that swap amounts using the "max" button do not receive the "transfer exceeded balance" error. It's difficult to do this correctly when we don't have the swap transaction, have opted to go with a multiple of a simple transfer fee for now. This means that users using the "max" button will be much less likely to see their swap rejected.

### Other changes

Since we are not using escrow invites anymore, rather than make the fee calculation function more complicated I opted to break the invite fee estimation....

### Tested

Made a successful swap using the max button

### How others should test

As above

### Related issues

- Fixes RET-495

### Backwards compatibility

Yes

jeanregisser reviewed (approved):
Hum tricky one, [because correctly estimating requires having the TX](https://ethereum.stackexchange.com/questions/121577/how-does-web3-js-estimate-gas-required-for-a-tx-using-the-estimategas-method).
The current approach will very likely be quite incorrect because swaps are more involved than simple transfers and require more gas.

More generally, the way the app currently estimates, then displays and send transactions is misleading for users.
The fee that they see and approve may change once they hit the send button.
We should rework this.

We can still merge this, as it's better than nothing. But we should add proper warnings/todo around this and say it needs more work.

kathaypacific commented:
@jeanregisser I added what we discussed and did some testing, seems to work well. We already have some custom fee estimation workaround for the invite escrow flow, rather than work around it I removed the workaround - so it will not work well anymore but it is a flow that we are not using and will remove very soon...

dievazqu commented on src/fees/saga.ts:
```diff
@@ -162,11 +187,7 @@ export function* estimateInviteFee(tokenAddress: string) {
-    tx.txo,
-    new BigNumber(STATIC_ESCROW_TRANSFER_GAS_ESTIMATE)
-  )
+  const feeInfo: FeeInfo = yield call(calculateFeeForTx, tx.txo)
```
Hmm, probably you already thought about this, but it seems that we are calculating a different fee for `estimateInviteFee`. I'm wondering if this could break something... 🤔

kathaypacific replied:
lol yeah i totally broke this but this is fee estimation for sending to escrow which is no longer supported...i thought about making `calculateFeeForTx` accept a static fee for addition AND a fee multiplier, but figured it's not worth the ugly function signature and opted to break this unused flow instead. i have a ticket in the backlog to remove escrow from the wallet

dievazqu reviewed (approved):
LGTM

kathaypacific commented on src/fees/saga.ts:
```diff
@@ -150,6 +155,26 @@ export function* estimateSendFee(tokenAddress: string) {
+    PLACEHOLDER_COMMENT
+  )
+
+  // TODO: calculate the fee accurately.
```
how would that work? i thought you need a buy or sell amount to query the `approveSwap` endpoint, and we can't get the sell amount without calculating the fee first 🙈

dievazqu replied:
I was thinking in something like, calling the approveSwap endpoint with the max amount as sell amount, but just to have an estimation on the gas. Then we have to query `approveSwap` again with the max amount - fees.
I don't like this option though, since it takes a relevant amount of time to query 0x.
